rpc/core/status.go

```diff
@@ -72,12 +76,26 @@ func Status() (*ctypes.ResultStatus, error) {
 
 	latestBlockTime := time.Unix(0, latestBlockTimeNano)
 
-	return &ctypes.ResultStatus{
+	result := &ctypes.ResultStatus{
 		NodeInfo:          p2pSwitch.NodeInfo(),
 		PubKey:            pubKey,
 		LatestBlockHash:   latestBlockHash,
 		LatestAppHash:     latestAppHash,
 		LatestBlockHeight: latestHeight,
 		LatestBlockTime:   latestBlockTime,
-		Syncing:           consensusReactor.FastSync()}, nil
+		Syncing:           consensusReactor.FastSync(),
+	}
+
+	// add ValidatorStatus if node is a validator
+	_, vals := consensusState.GetValidators()
```

Comment:
I think this is actually racey since the height may have incremented by the time we get here in which case the validator set here may not correspond to the Height we're returning. So I think we need to fetch the validators for the given height to be sure.

Reply:
see the new implementation https://github.com/tendermint/tendermint/blob/ed782e7508f430c2ece4fb2ebce82e2d4c2a6623/rpc/core/status.go#L100-L125